File: Python/2. Add Two Numbers.py
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        cur = res = ListNode()
        carry = 0
        while l1 or l2 or carry:
            if l1:
                carry += l1.val
                l1 = l1.next
            if l2:
                carry += l2.val
                l2 = l2.next
            carry, value = carry // 10, carry % 10
            cur.next = ListNode(value)
            cur = cur.next
        return res.next

File: 2. Add Two Numbers.js
var addTwoNumbers = function(l1, l2) {
    let cur = new ListNode();
        let res = cur;
        let carry = 0;
        while (l1 || l2 || carry) {
            if (l1) {
                carry += l1.val;
                l1 = l1.next;
            }
            if (l2) {
                carry += l2.val;
                l2 = l2.next;
            }
            let [div, mod] = [Math.floor(carry / 10), carry % 10];
            carry = div;
            cur.next = new ListNode(mod);
            cur = cur.next;
        }
        return res.next;
};